functions/Export-DbaAvailabiltyGroup.ps1

```diff
@@ -0,0 +1,173 @@
+
+.EXAMPLE
+Export-DbaAvailabilityGroup -SqlServer sql2014 -FilePath 'C:\temp\availability_group_exports' -NoClobber
+
+Exports all Availability Groups from SQL server "sql2014". Output scripts are witten to the C:\temp\availability_group_exports directory. If the export file already exists it will not be overwritten.
+
+.NOTES 
+Author: Chris Sommer (@cjsommer), cjsommmer.com
+
+.LINK 
+https://dbatools.io/Export-DbaAvailabilityGroup
+
+#>
+    [CmdletBinding(SupportsShouldProcess = $true)]
+	Param (
+        [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		    [Alias("ServerInstance", "SqlInstance")]
+		    [object[]]$SqlServer,
+
+		[Alias("OutputLocation", "Path")]
+		    [string]$FilePath,
+
+        [switch]$NoClobber ,
+
+		[object]$SqlCredential
+	)
+
+	DynamicParam { if ($sqlserver) { return Get-ParamSqlAvailabilityGroups -SqlServer $sqlserver[0] -SqlCredential $SqlCredential } }
+	
+    BEGIN
+    {
+        Write-Output "Beginning Export-DbaAvailabilityGroup"
+        $AvailabilityGroups = $PSBoundParameters.AvailabilityGroups
+    }
+
+    PROCESS
+    {
+        $SQLObj = New-Object "Microsoft.SqlServer.Management.Smo.Server" $SQLServer
+        $SQLObj.ConnectionContext.Connect()
+
+        # Get all of the Availability Groups and filter if required
+        $AllAGs =  $SQLObj.AvailabilityGroups
+
+        if ($AvailabilityGroups) {
+            Write-Verbose 'Filtering AvailabilityGroups'
+            $AllAGs = $AllAGs | Where-Object {$_.name -in $AvailabilityGroups}
+        }
+
+        if ($AllAGs.count -eq 0) {
+            Write-Output "No Availability Groups detected on '$SqlServer'"
+        }
+
+        # Set and create the OutputLocation if it doesn't exist
+        $SQLINST = $SQLServer.Replace('\','$')
+        $OutputLocation = "${FilePath}\${SQLINST}"
+
+        if (!(Test-Path $OutputLocation -PathType Container)) {
+            New-Item -Path $OutputLocation -ItemType Directory -Force | Out-Null
+        } 
+
+        # Script each Availability Group
+        foreach ($ag in $AllAGs) {
+            $AGName = $ag.Name
+
+            # Set the outfile name
+            if ($AppendDateToOutputFilename.IsPresent) {
+                $Dttm = (Get-Date -Format 'yyyyMMdd_hhmm')
+                $OutFile = "${OutputLocation}\${AGname}_${Dttm}.sql"
+            } else {
+                $OutFile = "${OutputLocation}\${AGname}.sql"
+            }
+
+            # Check NoClobber and script out the AG
+            if ($NoClobber.IsPresent -and (Test-Path -Path $OutFile -PathType Leaf)) {
+                Write-Warning "OutputFile '$OutFile' already exists. Skipping due to -NoClobber parameter"
+            } else {
+                Write-output "Scripting Availability Group [$AGName] on [$SQLServer] to '$OutFile'"
+
+                # Create comment block header for AG script
+                "/*" | Out-File -FilePath $OutFile -Encoding ASCII -Force
```

Comment:
PowerShell is particularly awesome when handling strings. You can do the following instead of line by line

"/*
blah blah
blah blah etc
Whatever
*/"  | Out-File -FilePath $OutFile -Encoding ASCII -Force

Please do, it's pretty :D Sorry I couldn't format it -- GitHub was hatin